sros2/test/sros2/commands/security/verbs/test_create_key.py

```diff
@@ -0,0 +1,111 @@
+# Copyright 2019 Canonical Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import tempfile
+from xml.etree import ElementTree
+
+from cryptography import x509
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import serialization
+from cryptography.hazmat.primitives.asymmetric import ec
+
+from ros2cli import cli
+
+from sros2.api import create_keystore
+
+
+def load_cert(path):
+    with open(path, 'rb') as f:
+        pem_data = f.read()
+    return x509.load_pem_x509_certificate(pem_data, default_backend())
+
+
+def load_csr(path):
+    with open(path, 'rb') as f:
+        pem_data = f.read()
+    return x509.load_pem_x509_csr(pem_data, default_backend())
+
+
+def load_private_key(path):
+    with open(path, 'rb') as f:
+        pem_data = f.read()
+    return serialization.load_pem_private_key(pem_data, password=None, backend=default_backend())
+
+
+def check_common_name(entity, expected_value):
+    names = entity.get_attributes_for_oid(x509.oid.NameOID.COMMON_NAME)
+    assert len(names) == 1
+    assert names[0].value == expected_value
+
+
+def check_cert_pem(path):
+    cert = load_cert(path)
+    check_common_name(cert.subject, u'/test_node')
+    check_common_name(cert.issuer, u'sros2testCA')
```

Comment:
I'd like to see the signature verified as well as checking the issuer name, something like...

```python3
def verify_signature(cert, signatory):
    signatory.public_key().verify(
        cert.signature,
        cert.tbs_certificate_bytes,
        ec.ECDSA(cert.signature_hash_algorithm),
    )

# ...

try:
    verify_signature(
        load_cert(os.path.join(keystore_dir, 'test_node', 'cert.pem')),
        load_cert(os.path.join(keystore_dir, 'ca.cert.pem')))
except exceptions.InvalidSignature:
        pytest.fail('Expected signature to be valid')
```